app/javascript/react/components/ReviewForm.js

```diff
@@ -0,0 +1,106 @@
+import { post } from 'fetch-mock' //What dis
+import React, { useState } from 'react'
+
+const ReviewForm = (props) => {
+  const [getReviewData, setReviewData] = useState([])
+  const [getNewReview, setNewReview] = useState({masks_customers: false, masks_employees: false, social_distancing: false, party_size: false })
+
+  const handleInputChange = event => {
+
+    const target = event.currentTarget;
+    const value = target.type === 'checkbox' ? target.checked : Number(target.value);
+
+    setNewReview({
+      ...getNewReview,
+      [event.currentTarget.name]: value
+    })
+  }
+
+  let handleSubmit = (event) => {
+    
+    event.preventDefault()
+    let formPayload = {
+      review: getNewReview
+    }
+
+    fetch(`/api/v1/restaurants/${props.id}/reviews`, {
+      credentials: "same-origin",
+      method: "POST",
+      body: JSON.stringify(formPayload),
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json"
+      }
+  })
+
+  .then(response => {
+    if (response.ok) {
+      return response
+  } else {
+      let errorMessage = `${response.status} (${response.statusText})`,
+      error = new Error(errorMessage)
+      throw error
+    }
+  })
+
+  .then(response => response.json())
+  .then(body => {
+    setReviewData([...getReviewData, getNewReview])
+  })
+  .catch(error => console.error(`Error in fetch: ${error.message}`))
+
+  }
```

Comment:
I see what you are going for by breaking apart the different phases of the response object.  Since we are still chaining onto the original fetch call with `.then`'s we will want to make that clear by eliminating the empty lines on 35, 45, and 51